src/features/game/lib/gameMachine.ts

```diff
@@ -201,7 +206,7 @@ export function startGame(authContext: Options) {
             },
             onDone: [
               {
-                target: "playing",
+                target: "loaded",
```

Comment:
An additional state may cause some side effects on our UI and adds additional overhead that we need to maintain. What is the purpose of creating an additional state in this case?

I would suggest using the `cond` to conditionally enter the correct state and abstracting the action so we don't duplicate ourselves.

See below where we don't need the `loaded` state

```
            onDone: [
              [
                {
                  target: "announcing",
                 // an imported function from a lib file
                  cond: hasAnnouncements,
                  // a reusable action
                  actions: "assignGame"
                },
                {
                  target: "playing",
                  actions: "assignGame"
                },
              ]
            ],
```

Down in the `actions`, we can define a reusable action:

E.g.
```
    {
      actions: {
        assignErrorMessage: assign<Context, any>({
          errorCode: (_context, event) => event.data.message,
          actions: [],
        }),
        assignGame: assign<Context, any>({
          state: (_, event) => event.data.state,
          onChain: (_, event) => event.data.onChain,
          owner: (_, event) => event.data.owner,
          offset: (_, event) => event.data.offset,
          sessionId: (_, event) => event.data.sessionId,
          fingerprint: (_, event) => event.data.fingerprint,
          itemsMintedAt: (_, event) => event.data.itemsMintedAt,
        }),
      },
    }
```